src/util/fields.py

```diff
@@ -1079,9 +1081,27 @@ def _update_field_usage_multi(cr, models, old, new, domain_adapter=None, skip_in
         """
         cr.execute(q.format(col_prefix=col_prefix), p)
 
-        # ir.exports.line
+        # ir.exports.line, base_import.mapping # noqa
         if only_models:
-            cr.execute(
+
+            def fix_paths(query, update_query, old_field_name):
+                cr.execute(query)
+
+                fixed_paths = {}
+                for record_id, related_model, path in cr.fetchall():
+                    new_path = [
+                        new
+                        if field.field_name == old_field_name and field.field_model in only_models
+                        else field.field_name
+                        for field in resolve_model_fields_path(cr, related_model, path)
+                    ]
+                    if len(new_path) == len(path) and new_path != path:
+                        fixed_paths[record_id] = "/".join(new_path)
+
+                if fixed_paths:
+                    cr.execute(update_query, [Json(fixed_paths), tuple(fixed_paths)])
```

Comment:
> i tried following the same idea here: inner function is ignorant of query parameters (assuming table and field name also as parameters)

You need to pass also `new_field_name` (instead of reusing `new`). Note that this would probably result in shorter code if you pass the table and column to update and build the update query internally. I'll let you decide about which version seems better if you implement and test them locally.